### PR TITLE
{Telemetry} Add platform information and some config values

### PR DESCRIPTION
**Pull request by fengzhou-msft**
**Description**  

Currently, we only have platform system name in telemetry, i.e., `windows`, `linux`, `macos`. This PR adds more platform details in a new property in telemetry, e.g., `windows-10-10.0.19041-sp0`, `Linux-5.0.0-1027-azure-x86_64-with-Ubuntu-18.04-bionic`.

This PR also adds all keys in `az config set/unset` commands in telemetry. For `az config set`, if the section or key is in the allow list (mainly for switch type configs), values will also be recorded. We can better analyze users' preference with this data.

 
**Testing Guide**  

1. Run the following command and config keys and values are recorded.
   `az config set auto-upgrade.enable=no extension.run_after_dynamic_install=no`

    > "Context.Default.AzureCLI.debug_info":"ConfigSet: auto-upgrade.enable=no extension.run_after_dynamic_install=no"

2. Run the following command and only config keys are recorded.
    `az config set defaults.group=myGroup storage.account=myAccount`

    > "Context.Default.AzureCLI.debug_info":"ConfigSet: defaults.group=*** storage.account=***"

3. Run the following command and keys are recorded.
    `az config unset auto-upgrade.enable defaults.group`

    > "Context.Default.AzureCLI.debug_info":"ConfigUnset: auto-upgrade.enable defaults.group"

4. A new property `Context.Default.VS.Core.OS.Platform` is recorded for all comamnds.

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
allow list is a global whitelist setting maintained by us, right?

**Comment by fengzhou-msft** [comment]
> allow list is a global whitelist setting maintained by us, right?

Yes. If the config value does not contain user-specific information, we can put it in the allow list to gain more insights on users' config preference.

File: src/azure-cli/azure/cli/command_modules/config/custom.py [hunk 1]
```diff
@@ -36,6 +44,12 @@ def config_set(cmd, key_value=None, local=False):
                 name = parts[1]
 
                 cmd.cli_ctx.config.set_value(section, name, _normalize_config_value(value))
+                if section in telemetry_allowed_sections_or_keys or key in telemetry_allowed_sections_or_keys:
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/config/custom.py:47]: I would suggest to put this filtering logic into telemetry module to make `config_set` as simple as possible.

Comment (fengzhou-msft) [reply src/azure-cli/azure/cli/command_modules/config/custom.py:47]: As this extra `debug_info` is command specific and in the future we may add other data for other modules. Putting it in telemetry module may need extra handling of each command module and less flexible.

**Comment by yungezz** [comment]
hi @houk-ms could you pls review the PR thanks.

File: src/azure-cli/azure/cli/command_modules/config/custom.py [hunk 2]
```diff
@@ -86,6 +91,7 @@ def config_unset(cmd, key=None, local=False):
 
         with ScopedConfig(cmd.cli_ctx.config, local):
             cmd.cli_ctx.config.remove_option(section, name)
+    telemetry.set_debug_info('ConfigUnset', ' '.join(key))
```
Comment (houk-ms) [inline src/azure-cli/azure/cli/command_modules/config/custom.py:94]: If user unset a property, why not just remove the property from Telemetry？

**Review by houk-ms — APPROVED** [review]

**Review by jiasli — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/config/custom.py (hunk 2 above)
Comment (fengzhou-msft) [reply src/azure-cli/azure/cli/command_modules/config/custom.py:94]: To record user behavior of the unset command.

**Pull request merged**